AdColony/src/main/java/com/mopub/mobileads/AdColonyRewardListenersDispatcher.java

```diff
@@ -0,0 +1,78 @@
+package com.mopub.mobileads;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.adcolony.sdk.AdColony;
+import com.adcolony.sdk.AdColonyReward;
+import com.adcolony.sdk.AdColonyRewardListener;
+
+import java.lang.ref.WeakReference;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+class AdColonyRewardListenersDispatcher implements AdColonyRewardListener {
+
+    private static AdColonyRewardListenersDispatcher instance;
+    private final List<WeakReference<AdColonyZoneSpecificRewardListener>> mListeners = new ArrayList<>();
+
+    static AdColonyRewardListenersDispatcher getInstance() {
+        if (instance == null) {
+            synchronized (AdColonyRewardListenersDispatcher.class) {
+                if (instance == null) {
+                    instance = new AdColonyRewardListenersDispatcher();
+                }
+            }
+        }
+        return instance;
+    }
+
+    void setGlobalRewardListener() {
```

Comment:
Setters usually have a value, in this case, to indicate what the value is, I would rename this method to setAsGlobalRewardListener or setupGlobalRewardListener

Reply:
Fixed